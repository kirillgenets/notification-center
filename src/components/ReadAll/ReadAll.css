/* ReadAll */

.button {
  width: 100%;
  padding: 23px 0;
  border: none;

  background-color: #51A8E7;

  text-transform: uppercase;
  color: #ffffff;

  cursor: pointer;
  transition: 0.3s;
}

.button:hover {
  opacity: 0.8;
}

.button:active {
  opacity: 0.6;
}