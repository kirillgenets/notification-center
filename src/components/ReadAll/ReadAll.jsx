import React from 'react';
import PropTypes from 'prop-types';
import styles from './ReadAll.css';

const ReadAll = (props) => {
  const handleButtonClick = () => {
    props.markAllNotificationsAsRead(props.notifications);
  };

  return (
    <button
      className={styles.button}
      onClick={handleButtonClick}
    >
        Mark all as read
    </button>
  );
};

ReadAll.propTypes = {
  markAllNotificationsAsRead: PropTypes.func.isRequired,
  notifications: PropTypes.array.isRequired,
};

export default ReadAll;
