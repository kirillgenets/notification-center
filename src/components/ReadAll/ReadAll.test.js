import React from 'react';
import renderer from 'react-test-renderer';
import ReadAll from './ReadAll';

it('ReadAll renders correctly', () => {
  const tree = renderer
    .create(
      <ReadAll />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
