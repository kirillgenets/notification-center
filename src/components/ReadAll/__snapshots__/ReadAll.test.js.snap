// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadAll renders correctly 1`] = `
<button
  onClick={[Function]}
>
  Mark all as read
</button>
`;
